### PR TITLE
Build system

### PR DESCRIPTION
**Pull request by dryajov**

### CODE REVIEW

**Comment by dryajov** [comment]
@elcritch as discussed during the last call, this moves the build system from nimble to nimbus build system. Lets finalize this  and merge it with the work you've already done in #65.

File: libp2pdht/private/eth/p2p/discoveryv5/protocol.nim [hunk 1]
```diff
@@ -1032,6 +1032,76 @@ func init*(
     bitsPerHop: bitsPerHop
   )
 
+proc newProtocol*(
```
Comment (dryajov) [inline libp2pdht/private/eth/p2p/discoveryv5/protocol.nim:1035]: @elcritch this is the removed code from master.

Comment (elcritch) [reply libp2pdht/private/eth/p2p/discoveryv5/protocol.nim:1035]: Ah yes, I'll see if I can merge it with @yyoncho's changes

**Comment by elcritch** [comment]
I updated and tweaked this a bit. It should be able to be merged with the nimble lock files. 

I updated the deps and noticed that there's a few deps which don't appear to be in the submodules. I'm not sure how that's working:

```sh
?? vendor/dnsclient.nim/
?? vendor/nim-protobuf-serialization/
?? vendor/nim-testutils/
?? vendor/nim-zlib/
?? vendor/npeg/
?? vendor/packages/
```

**Comment by elcritch** [comment]
Closing in favor of: https://github.com/codex-storage/nim-codex-dht/pull/67. We can always go back to a full submodule setup later. It's only a minor tweak to the makefile. 

Hopefully this should avoid the issue of leaf-deps in Nimbus also having a vendored deps.

**Pull request closed without merging**